18759.yml

```diff
@@ -5,44 +5,36 @@ on:
     branches: ["submission-*"]
 
 jobs:
-  repo_check:
+  extract_url:
     runs-on: ubuntu-latest
     outputs:
       outcome: ${{steps.get-submission-url.outcome}}
       submissionUrl: ${{steps.get-submission-url.outputs.result}}
     steps:
       - name: Check out the parent repository with student submission data
         uses: actions/checkout@v2
-      - name: Extract the student submission repo from URL
+      - name: Extract the student submission URL
         uses: actions/github-script@v5
         id: get-submission-url
         continue-on-error: true
         with:
           script: |
             const submission = require('./submission.json')
-            const herokuURL =  submission.checklist[0].result
-            if ((herokuURL.split("/").length >= 2) && herokuURL.includes("herokuapp.com")) {
-              return herokuURL
-            } else {
-              throw "The submitted URL seems to be invalid"
-            }
+            return submission.checklist[0].result
           result-encoding: string
       - name: Report invalid repository URL in submission
         if: steps.get-submission-url.outcome != 'success'
         uses: pupilfirst/actions/grading@v1
         with:
           fail_submission: true
-          feedback: The submitted URL is either unavailable or not a valid Heroku app. Please make sure that you submit a valid Heroku application URL.
+          feedback: We failed to extract the URL for your live web application from your submission. Please submit again; if this issue persists, please reach out to our team for support.
```

Comment:
This won't be required anymore as the `URL` as `get-submission-url` is no longer throwing an error.

Reply:
That's just a catch for if the above script execution fails for any reason.

Reply:
@harigopal The chances of that script failing are super low. 

Possibilities 🤔 
1. Failed to load `submission.json`. The action will fail to send the status as the submission ID is required for reporting/reviewing
2. Checklist is missing in the `submission.json`; In this case, the action should fail as it would be a bug on our end
3. Invalid URL; This won't be a failure, but our submission builder does a minimum URL validation. so safe to ignore

This was explicitly added to reject submissions with junk links. (which is no longer valid)